### PR TITLE
DEPR: add shims for util + TST: test that we work in downstream packages

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

**Comment by jreback** [comment]
should update to have at least an example for each one.

@jorisvandenbossche @TomAugspurger

File: pandas/util/decorators.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import warnings
+
+warnings.warn("pandas.util.decorators is deprecated and will be "
+              "removed in a future version, import "
+              "from pandas.util",
+              DeprecationWarning, stacklevel=3)
+
+from pandas.util import cache_readonly  # noqa
```
Comment (jorisvandenbossche) [inline pandas/util/decorators.py]: I don't know of anybody using it, but maybe also Appender and Substition to be sure? (or import *)

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

Nice to add such test

**Comment by TomAugspurger** [comment]
@jreback I'm taking care of the release note for this in my doc updates (will have a PR shortly).

**Comment by jreback** [comment]
thanks @TomAugspurger yeah will merge when this is fixed.

**Comment by TomAugspurger** [comment]
@jreback https://travis-ci.org/pandas-dev/pandas/jobs/229097977 failed. Does that error message make sense to you?

**Comment by jreback** [comment]
@TomAugspurger still working on things.

**Comment by jreback** [comment]
turns out our build-test build was not actually testing the correct installation :<

**Comment by jreback** [comment]
lon it built correctly but still some issue
will investigate

**Comment by jorisvandenbossche** [comment]
This all passed. Good to merge then?

**Comment by TomAugspurger** [comment]
The build one failed https://travis-ci.org/pandas-dev/pandas/jobs/229133755

**Comment by jreback** [comment]
not yet

**Comment by jorisvandenbossche** [comment]
Ah, yes, I didn't look at the allowed failures...

**Comment by jorisvandenbossche** [comment]
But, it is only the setting up of the test build env, not the actual code changes? Those could already be merged?

**Comment by TomAugspurger** [comment]
Ah, yes I think so. Although, those tests will be skipped if the deps aren't around. Do we we have dask and geopandas anywhere else in the matrix?

**Comment by jorisvandenbossche** [comment]
we can test those two locally

**Comment by jorisvandenbossche** [comment]
I checked out this PR, and can confirm it fixed the geopandas import for me

**Comment by TomAugspurger** [comment]
I think we're good then. I'll ensure they work after tagging but before
uploading.

**Comment by jorisvandenbossche** [comment]
@jreback OK to merge and fix the test later? 
(I can also split off just the code changes to merge only that)

**Comment by jreback** [comment]
yes that is fine

**Pull request merged**